src/v/model/model.cc

```diff
@@ -236,11 +236,11 @@ std::ostream& operator<<(std::ostream& o, const broker_shard& bs) {
 std::ostream& operator<<(std::ostream& o, compaction_strategy c) {
     switch (c) {
     case compaction_strategy::offset:
-        return o << "{compaction_strategy::offset}";
+        return o << "offset";
     case compaction_strategy::timestamp:
-        return o << "{compaction_strategy::timestamp}";
+        return o << "timestamp";
     case compaction_strategy::header:
-        return o << "{compaction_strategy::header}";
+        return o << "header";
     }
     return o << "{unknown model::compaction_strategy}";
```

Comment:
```suggestion
    __builtin_unreachable();
```

since `compaction_strategey` is an enum class the compiler knows that all of the cases are covered in the switch statement. adding `__builtin_unreachable();` is only necessary to avoid a warning on gcc which sometimes doesn't bother to be as strict as clang.